### PR TITLE
cmd/clone: check clone path

### PR DESCRIPTION
**Pull request by zhijian-pro**

### CODE REVIEW

File: cmd/clone.go [hunk 1]
```diff
@@ -127,3 +144,20 @@ func clone(ctx *cli.Context) error {
...
 	return nil
 }
+func findMountpoint(dir string) (string, error) {
+	odir := dir
+	for dir != "" {
+		inode, err := utils.GetFileInode(dir)
+		if err != nil {
+			return "", fmt.Errorf("get inode of %s: %s", dir, err)
+		}
+		if inode == 1 {
```
Comment (SandyXSD) [inline cmd/clone.go:154]: meta.RootInode

File: cmd/clone.go [hunk 2]
```diff
@@ -127,3 +144,20 @@ func clone(ctx *cli.Context) error {
 	}
 	return nil
 }
+func findMountpoint(dir string) (string, error) {
```
Comment (SandyXSD) [inline cmd/clone.go:147]: unify the implementation for other commands (warmup, rmr, openController, etc)

File: cmd/clone.go [hunk 3]
```diff
@@ -84,6 +84,23 @@ func clone(ctx *cli.Context) error {
 	if err != nil {
 		return fmt.Errorf("abs of %s: %s", dst, err)
 	}
+
+	srcMp, err := findMountpoint(srcAbsPath)
+	if err != nil {
+		return err
+	}
+	dstMp, err := findMountpoint(filepath.Dir(dstAbsPath))
```
Comment (SandyXSD) [inline cmd/clone.go:92]: no need, just check if dstAbsPath has srcMp as the prefix

Comment (zhijian-pro) [reply cmd/clone.go:92]: It doesn't apply in this case
```
src=/tmp/mp1/dir1
dst=/tmp/mp10/dir2
```

File: cmd/clone.go [hunk 4]
```diff
@@ -84,6 +83,23 @@ func clone(ctx *cli.Context) error {
...
+	if err != nil {
+		return err
+	}
+	if srcMp != dstMp {
+		return fmt.Errorf("the clone DST path should be at the same mount point as the SRC path")
+	}
+
+	if strings.HasPrefix(strings.TrimLeft(dstAbsPath, dstMp), strings.TrimLeft(srcAbsPath, srcMp)) {
```
Comment (davies) [inline cmd/clone.go:99]: trim is not needed

Comment (davies) [reply cmd/clone.go:99]: a  -> ab

**Review by SandyXSD — APPROVED** [review]

**Pull request merged**